Application/Published/Upgrade-AdminApp/Deploy-EdFiOdsAdminApp.ps1

```diff
@@ -0,0 +1,220 @@
+.PARAMETER ResourceGroupLocation
+    Existing resource group location
+.PARAMETER AdminAppName
+    A friendly name to help identify AdminApp within Azure.  Must be no more than 64 characters long
+.PARAMETER AppInsightLocation
+	Existing app insight location, mostly same as resouce group location
+.PARAMETER ProductionApiUrl
+	Existing ODS/ API Url that Admin App can connect
+.PARAMETER EncryptionKey
+	Base64-encoded 256 bit key appropriate for use with AES encryption
+.PARAMETER TemplateFileDirectory
+	Points the script to the directory that holds the Ed-Fi ODS install templates.  By default that directory is the same as the one that contains this script.
+.PARAMETER AdminAppVersion
+	Admin app version to be deployed
+.PARAMETER Edition
+	Edition (Test, Release) of the Azure Deploy scripts to deploy.  Defaults to Release.
+.EXAMPLE
+	.\Deploy-EdFiOds.ps1 -ResourceGroupName "Ed-Fi-Ods-Resourcegroup" -ResourceGroupLocation "South Central US" -AdminAppName "AdminApp-Latest" -AppInsightLocation "South Central US" -ProductionApiUrl "https://edfiodsapiwebsite-production-yuw8iui32.azurewebsites.net"
+    Deploys the provided version of the AdminApp to the South Central US Azure region with the default instance name
+#>
+Param(
+
+	[string] 
+	[Parameter(Mandatory=$true)] 
+	$ResourceGroupName,
+
+	[ValidateSet("East US", "South Central US")]
+	[string] 
+	[Parameter(Mandatory=$true)]	
+	$ResourceGroupLocation,	
+
+	[string] 
+	[Parameter(Mandatory=$true)] 
+	$AdminAppName,
+
+	[ValidateSet("East US", "South Central US")]
+	[string] 
+	[Parameter(Mandatory=$true)]	
+	$AppInsightLocation,
+	
+	[string]
+	[Parameter(Mandatory=$true)]
+	$ProductionApiUrl,
+
+	[string]
+	[Parameter(Mandatory=$false)] 
+	$EncryptionKey,
+
+	[string] 
+	[Parameter(Mandatory=$false)] 
+	$TemplateFileDirectory = '$PSScriptRoot',
+
+	[string]
+	[Parameter(Mandatory=$false)]
+	$AdminAppVersion = '2.2.0',
+	
+	[string]
+	[ValidateSet('Release','Test')]
+	$Edition = 'Release'	
+)
+
+Import-Module $PSScriptRoot\..\Dependencies.psm1 -Force -DisableNameChecking
+Import-Module $PSScriptRoot\..\EdFiOdsDeploy.psm1 -Force -DisableNameChecking
+
+Use-Module "AzureRM" "4.3.1"
+Use-Module "AzureRM.profile" "3.3.1"
+
+# Powershell doesn't support using $PSScriptRoot in parameter defaults,
+# so using that value as a marker to update with the real PSScriptRoot value
+if ($TemplateFileDirectory -eq '$PSScriptRoot')
+{
+	$TemplateFileDirectory = $PSScriptRoot;
+}
+
+$CacheTimeOut = "10"
+
+$AdminAppTemplateFile = "$TemplateFileDirectory\OdsAdminApp.Upgrade.json"
+$AdminAppTemplateParametersFile = "$TemplateFileDirectory\OdsAdminApp.parameters.Upgrade.json"
+
+function Get-SqlServerInfo()
+{	
+	$hostName = Read-Host -Prompt "SQL Server Hostname (ex: sql.mydomain.com,1433)"
+	$hostName = $hostName.Replace(':', ',')
+	$adminCredentialsPrompt = "Please enter a username and password for your SQL Server.  These credentials will be used to create new database users for your Ed-Fi ODS installation."
```

Comment:
Same question here - why not provide as function parameters?

If this is an existing pattern then I'm not trying to change it right now. But I don't know the existing pattern, and I'm just reading this script from top to bottom. Seems strange to have a mix of items passed in as function parameters and those that are prompted from the user at the command line.